core/templates/dev/head/collection_editor/editor_tab/CollectionLinearizerServiceSpec.js

```diff
@@ -0,0 +1,204 @@
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Tests for CollectionLinearizerService.
+ */
+
+describe('Collection linearizer service', function() {
+  var CollectionObjectFactory = null;
+  var CollectionNodeObjectFactory = null;
+  var CollectionLinearizerService = null;
+
+  var firstCollectionNode = null;
+  var secondCollectionNode = null;
+  var thirdCollectionNode = null;
+
+  beforeEach(module('oppia'));
+
+  beforeEach(inject(function($injector) {
+    CollectionObjectFactory = $injector.get('CollectionObjectFactory');
+    CollectionNodeObjectFactory = $injector.get('CollectionNodeObjectFactory');
+    CollectionLinearizerService = $injector.get('CollectionLinearizerService');
+
+    var firstCollectionNodeBackendObject = {
+      exploration_id: 'exp_id0',
+      prerequisite_skills: [],
+      acquired_skills: ['exp title0'],
+      exploration_summary: {
+        title: 'exp title0',
+        category: 'exp category',
+        objective: 'exp objective'
+      }
+    };
+    firstCollectionNode = CollectionNodeObjectFactory.create(
+        firstCollectionNodeBackendObject);
+
+    var secondCollectionNodeBackendObject = {
+      exploration_id: 'exp_id1',
+      prerequisite_skills: ['exp title0'],
+      acquired_skills: ['exp title1'],
+      exploration_summary: {
+        title: 'exp title1',
+        category: 'exp category',
+        objective: 'exp objective'
+      }
+    };
+    secondCollectionNode = CollectionNodeObjectFactory.create(
+        secondCollectionNodeBackendObject);
+
+    var thirdCollectionNodeBackendObject = {
+      exploration_id: 'exp_id2',
+      prerequisite_skills: ['exp title1'],
+      acquired_skills: ['exp title2'],
+      exploration_summary: {
+        title: 'exp title2',
+        category: 'exp category',
+        objective: 'exp objective'
+      }
+    };
+    thirdCollectionNode = CollectionNodeObjectFactory.create(
+        thirdCollectionNodeBackendObject);
+  }));
+
+  var _createLinearCollection = function() {
+    var collection = CollectionObjectFactory.createEmptyCollection();
+    collection.addCollectionNode(firstCollectionNode);
+    collection.addCollectionNode(secondCollectionNode);
+    collection.addCollectionNode(thirdCollectionNode);
+    return collection;
+  };
+
+  describe('Collection linearizer node removal function', function() {
+    it('should correctly remove a node from a collection with a single node',
+        function() {
+          var collection = CollectionObjectFactory.createEmptyCollection();
+          collection.addCollectionNode(firstCollectionNode);
+          expect(collection.containsCollectionNode('exp_id0')).toBe(true);
+          CollectionLinearizerService.removeCollectionNode(
```

Comment:
Wait, I thought we agreed that I can't do: `return CollectionUpdateService.deleteCollectionNode(collection, explorationId);` since deleteCollectionNode doesn't return a boolean!!?